### PR TITLE
support count in word movement commands

### PR DESCRIPTION
**Pull request by zhaojiangbin**
word_forward, word_backward, word_end_forward

In other words, now you can do <kbd>3w</kbd>, <kbd>5b</kbd>, and <kbd>4e</kbd> in vi mode, for example.

### CODE REVIEW

File: lapce-data/src/buffer.rs [hunk 1]
```diff
@@ -1184,6 +1184,20 @@ impl Buffer {
...
     }
+
+    // The folder function calculates offset of the next/previous word boundary
+    // for the given offset.
+    fn move_by_word<F>(
+        &self, offset: usize, count: usize, tab_width: usize, folder: F
+    ) -> (usize, ColPosition)
+    where
+        F: FnMut(usize, ()) -> usize {
+        let new_offset = std::iter::repeat(()).take(count).fold(offset, folder);
```
Comment (bugadani) [inline lapce-data/src/buffer.rs:1195]: This is a bit cryptic to eyes not accustomed to functional primitives like fold. I think a comment might be helpful.

File: lapce-data/src/buffer.rs [hunk 2]
```diff
@@ -1184,6 +1184,20 @@ impl Buffer {
             line
         }
     }
+
+    // The folder function calculates offset of the next/previous word boundary
+    // for the given offset.
```
Comment (bugadani) [inline lapce-data/src/buffer.rs:1189]: Could you turn this into a proper doc comment, please?

**Review by bugadani — COMMENTED** [review]
Thanks for this!

I'm not sure what the plan is right now. There's a `refactor` branch that seems to touch a lot of editing related code, so I think @dzhou121 would want to hold off merging this PR until work on that branch completes?

**Comment by zhaojiangbin** [comment]
> Thanks for this!
> 
> I'm not sure what the plan is right now. There's a `refactor` branch that seems to touch a lot of editing related code, so I think @dzhou121 would want to hold off merging this PR until work on that branch completes?

I didn't know that. I don't mind waiting.

File: lapce-data/src/buffer.rs (hunk 1 above)
Comment (zhaojiangbin) [reply lapce-data/src/buffer.rs:1195]: Will try.

**Comment by zhaojiangbin** [comment]
@dzhou121, @bugadani, is this PR still being considered?

**Comment by bugadani** [comment]
Is there a way to add tests to this? Something like this one? (where did my cute test framework go? 😢 ) https://github.com/lapce/lapce/blob/f0f3dab62ceb4eaf4eca19ac2976227d46f9395a/lapce-core/src/buffer.rs#L1099

**Review by bugadani — COMMENTED** [review]
Thanks for the changes! I like the general direction (and I especially like how you placed the character offsets under the strings, that's very helpful!), but I still have a few readability concerns. Those are nothing serious, and I don't mind if you decide not to fix those.

**Comment by zhaojiangbin** [comment]
> Thanks for the changes! I like the general direction (and I especially like how you placed the character offsets under the strings, that's very helpful!), but I still have a few readability concerns. Those are nothing serious, and I don't mind if you decide not to fix those.

All good points. Fixed as much as I can.

**Review by bugadani — APPROVED** [review]
Thanks :)

**Pull request merged**